src/installer.ts

```diff
@@ -27,31 +27,125 @@ if (!tempDirectory) {
   tempDirectory = path.join(baseLocation, 'actions', 'temp');
 }
 
-export class DotnetCoreInstaller {
+/**
+ * Represents the inputted version information
+ */
+export class DotNetVersionInfo {
+  private fullversion: string;
+  private isExactVersionSet: boolean = false;
+
   constructor(version: string) {
-    if (semver.valid(semver.clean(version) || '') == null) {
-      throw 'Implicit version not permitted';
+    // Check for exact match
+    if (semver.valid(semver.clean(version) || '') != null) {
+      this.fullversion = semver.clean(version) as string;
+      this.isExactVersionSet = true;
+
+      return;
+    }
+
+    //Note: No support for previews when using generic

```

Comment:
Allowing preview versions can be added as a task once this is completed. The Azure DevOps version of this task has a flag which is used to find the latest preview version (if applicable).
https://github.com/microsoft/azure-pipelines-tasks/blob/master/Tasks/DotNetCoreInstallerV1/task.json#L56